apisix/init.lua

```diff
@@ -868,6 +869,54 @@ local function healthcheck_passive(api_ctx)
 end
 
 
+function _M.status()
+    core.response.exit(200, core.json.encode({
+        status = "ok" }),
+        { ["Content-Type"] = "application/json"
+    })
+end
+
+function _M.status_ready()
+    local local_conf = core.config.local_conf()
+    local provider = core.table.try_read_attr(local_conf, "deployment",
+                                              "role_traditional", "config_provider")
+    if provider == "yaml" or provider == "etcd" then
```

Comment:
The fact that you only use `role_traditional.config_provider` for assertions makes file-based standalone modes not work properly, so they always fall back to the default value of `etcd` for `role_traditional.config_provider`, which uses the `role_data_plane.config_provider`, which you should adapt.
Otherwise, this one status API can't simply be enabled in environments like Kubernetes probes, and we'd have to make special adaptations in the helm chart, which is neither reasonable nor sustainable.